src/UnitTests/Helpers/TestBaseClass.cs

```diff
@@ -7,14 +8,35 @@
 /// </summary>
 public class TestBaseClass : IEntryExitDecorator
 {
-    public void OnEntry()
+    public virtual void OnEntry()
     {
         // Ensure that every test has the InUnitTestRunner flag
         // set, so threading doesn't go nuts.
         Splat.ModeDetector.Current.SetInUnitTestRunner(true);
     }
 
-    public void OnExit()
+    public virtual void OnExit()
     {
     }
 }
+
+public class TempFileBaseClass : TestBaseClass
+{
+    public DirectoryInfo Directory { get; set; }
+
+    public override void OnEntry()
+    {
+        var f = Path.GetTempFileName();
+        var name = Path.GetFileNameWithoutExtension(f);
+        File.Delete(f);
+        Directory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), name));
+        Directory.Create();
+        base.OnEntry();
+    }
```

Comment:
Yeah, this one is stretching the unit test moniker. There's no way around it, the SimpleRepositoryModel is used in places where rothko is not yet loaded. And instantiated often enough that querying a service would impact performance if it were available, so we'd need to start caching services more aggressively to be safe using them in models.

We can move these to integration tests, I'm fine with that.